src/writer.rs

```diff
@@ -0,0 +1,58 @@
+use std::io::{Write, Error, ErrorKind};
+use std::io;
+use InflateStream;
+
+pub struct InflateWriter<W: Write> {
+    inflater: InflateStream,
+    writer: W
+}
+
+impl<W: Write> InflateWriter<W> {
+    pub fn new(w: W) -> InflateWriter<W> {
+        InflateWriter { inflater: InflateStream::new(), writer: w }
+    }
+
+    pub fn finish(mut self) -> io::Result<W> {
```

Comment:
If you look at the [source](https://docs.rs/flate2/0.2.17/src/flate2/deflate.rs.html#362) it's mut self in flate2 as well (It just doesn't show in the docs). I think it has to be mutable to compile.